src/renderer/opengl.rs

```diff
@@ -16,63 +17,202 @@ use glutin::{
 };
 use glutin_winit::DisplayBuilder;
 use raw_window_handle::HasRawWindowHandle;
-use winit::dpi::PhysicalSize;
+use skia_safe::{
+    canvas::Canvas,
+    gpu::{
+        backend_render_targets::make_gl, gl::FramebufferInfo, surfaces::wrap_backend_render_target,
+        DirectContext, SurfaceOrigin,
+    },
+    ColorType,
+};
 use winit::{
-    event_loop::EventLoop,
+    dpi::PhysicalSize,
+    event_loop::{EventLoop, EventLoopProxy},
     window::{Window, WindowBuilder},
 };
 
-pub struct GlWindow {
-    pub window: Window,
-    config: Config,
-}
+#[cfg(target_os = "windows")]
+pub use super::vsync::VSyncWinDwm;
+
+#[cfg(target_os = "macos")]
+pub use super::vsync::VSyncMacos;
+
+use super::{SkiaRenderer, VSync, WindowConfig, WindowConfigType};
 
-pub struct Context {
-    surface: Surface<WindowSurface>,
+use crate::{profiling::tracy_gpu_zone, window::UserEvent};
+
+#[cfg(feature = "gpu_profiling")]
+use crate::profiling::{opengl::create_opengl_gpu_context, GpuCtx};
+
+pub struct OpenGLSkiaRenderer {
+    // NOTE: The destruction order is important, so don't re-arrange
```

Comment:
Maybe a `SAFETY` could fit better in order to convey that a critical invariant is upheld here.

Reply:
I'll keep the `NOTE:` the syntax highlighter highligts it, but not `SAFETY:`